src/TableView.jl

```diff
@@ -1,119 +1,158 @@
 module TableView
-
-using WebIO
-using JSExpr
-using JuliaDB
-using DataValues
-
-import JuliaDB: DNDSparse, DNextTable, NextTable
-
-function JuliaDB.subtable(t::DNextTable, r)
-    table(collect(rows(t)[r]), pkey=t.pkey)
-end
-
-showna(xs) = xs
-function showna(xs::AbstractArray{T}) where {T<:DataValue}
-    map(xs) do x
-        isnull(x) ? "NA" : get(x)
+using Tables, TableTraits, IteratorInterfaceExtensions
+using WebIO, JSExpr, JSON, Dates
+using Observables: @map
+
+function showtable(x; dark = false)
+    if Tables.istable(typeof(x))
+        return _showtable(x, dark)
+    elseif TableTraits.isiterabletable(x)
+        it = IteratorInterfaceExtensions.getiterator(x)
+        return _showtable(Tables.DataValueUnwrapper(it), dark)
     end
+    throw(ArgumentError("Argument is not a table."))
 end
 
-function showna(xs::Columns)
-    rows(map(showna, columns(xs)))
+function _showtable(table, dark)
+    length(Tables.rows(table)) > 10_000 ? _showtable_async(table, dark) :
+                                          _showtable_sync(table, dark)
 end
 
-function showtable(t::Union{DNextTable, NextTable}; rows=1:100, colopts=Dict(), kwargs...)
-    w = Scope(imports=["https://cdnjs.cloudflare.com/ajax/libs/handsontable/0.34.0/handsontable.full.js",
-                       "https://cdnjs.cloudflare.com/ajax/libs/handsontable/0.34.0/handsontable.full.css"])
-
-    trunc_rows = max(1, first(rows)):min(length(t), last(rows))
-    subt = JuliaDB.subtable(t, trunc_rows)
-
-    headers = colnames(subt)
-    cols = [merge(Dict(:data=>n), get(colopts, n, Dict())) for n in headers]
+function _showtable_sync(table, dark)
+    schema = Tables.schema(table)
```

Comment:
There needs to be some code path that handles situation where no schema is available. I think this function returns ``nothing`` in that case, but I'm not 100% sure.

Reply:
Yeah, it does. I guess we should just not print a header in that case? Or is there any other mechanism to handle schemaless tables?

Reply:
Well, they are not really schema less, it is just that you don't know the schema until you start iterating. So the strategy would be to iterate the first element, look at the type of that and that should tell you what the column names and types are.